### PR TITLE
Generalize MGTwoLevelTransfer for simplex meshes

### PR DESCRIPTION
**Pull request by peterrum**
~Should be quite easy~. Introduce a second d-dimensional projection matrix:
https://github.com/dealii/dealii/blob/3679ade256b019723697e86847ebcbac70b1899f/include/deal.II/multigrid/mg_transfer_global_coarsening.h#L174-L177

and specialize the evaluation kernels just like in `FEEvaluation` for the general case.

Depends on ~#11426, #11428~, #11422, and ~#11436~.

### CODE REVIEW

**Comment by peterrum** [comment]
Once #11426, #11428, #11422, and #11436 are merged, this PR is ready and run global coarsening on simplex meshes (polynomial for 2D and 3D and geometric for 2D). The only relevant changes are in the files include/deal.II/multigrid/mg_transfer_global_coarsening.h and include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h.

**Comment by peterrum** [comment]
This PR is ready from my side!

**Review by kronbichler — APPROVED** [review]
Good to go once we rebase.

**Review by kronbichler — APPROVED** [review]

**Pull request merged**
